core/roboconf-dm-rest-services/src/main/java/net/roboconf/dm/rest/services/internal/resources/DebugResource.java

```diff
@@ -165,29 +212,95 @@ public Response checkMessagingConnectionForTheDm() {
 	 * #checkMessagingConnectionWithAgent(java.lang.String)
 	 */
 	@Override
-	public Response checkMessagingConnectionWithAgent( String rootInstanceName ) {
-		return Response.status( Status.SERVICE_UNAVAILABLE ).entity( "Not yet implemented." ).build();
+	public Response checkMessagingConnectionWithAgent( String applicationName,
+	                                                   String rootInstanceName,
+	                                                   String message,
+	                                                   long timeout ) {
+
+		if ( message == null ) {
+			// Generates an unique message.
+			message = UUID.randomUUID().toString();
+		}
+
+		// Max timeout
+		if ( timeout > MAXIMUM_TIMEOUT ) {
+			this.logger.warning( "Timeout " + timeout + "ms is above maximum limit " + MAXIMUM_TIMEOUT + "ms. Normalizing" );
+			timeout = MAXIMUM_TIMEOUT;
+		}
+
+		boolean hasReceived;
+		try {
+			// Ping the agent!
+			hasReceived = this.manager.pingAgent( applicationName, rootInstanceName, message, timeout );
+
+		} catch ( IOException e ) {
+			// Something bad has happened!
+			final String eMessage = "Unable to ping agent " + rootInstanceName + " with message " + message;
+			logger.log( Level.SEVERE, eMessage, e );
+			return Response.status( Status.INTERNAL_SERVER_ERROR ).entity( eMessage ).build();
+		} catch ( InterruptedException e ) {
+			// Something unexpected has happened!
+			final String eMessage = "Interrupted while waiting for ping response from agent " + rootInstanceName
+					+ " message " + message;
+			logger.log( Level.WARNING, eMessage, e );
+			return Response.status( Status.INTERNAL_SERVER_ERROR ).entity( eMessage ).build();
+		}
+
+		// Return result.
+		if ( hasReceived ) {
+			final String rMessage = "Has received ping response " + message + " from agent " + rootInstanceName;
+			logger.fine( rMessage );
+			return Response.status( Status.OK ).entity( rMessage ).build();
+		}
+		else {
+			final String rMessage = "Did not receive ping response " + message + " from agent " + rootInstanceName
+					+ " before timeout (" + timeout + "ms)";
+			logger.warning( rMessage );
+			return Response.status( 408 ).entity( rMessage ).build();
+		}
+
 	}
 
 
 	/* (non-Javadoc)
 	 * @see net.roboconf.dm.rest.services.internal.resources.IDebugResource
-	 * #diagnosticInstance(java.lang.String)
+	 * #diagnoseInstance(java.lang.String)
 	 */
 	@Override
-	public Response diagnosticInstance( String instancePath ) {
-		return Response.status( Status.SERVICE_UNAVAILABLE ).entity( "Not yet implemented." ).build();
+	public Response diagnoseInstance( String applicationName, String instancePath ) {
+		final Application application = this.manager.findApplicationByName( applicationName );
```

Comment:
With so little methods, you could only have one return statement.
This is more convenient for debug by limiting the number of breakpoints.